### ISSUE TITLE
S1172 false positive when function parameter is used in a tuple

### ISSUE THREAD

**Issue by volgunin** [SonarSource/sonar-dotnet#3307]
### Description

When a function parameter is used inside the function as a part of the tuple, S1172 is raised.

### Repro steps
```C#
        private class Test
        {
            private readonly List<(int X, int Y)> features = new List<(int X, int Y)>();

            public void TestMethod(int x, int y)
            {
                this.features.Add((x, y));
            }
        }
```
### Expected behavior

No warning.

### Actual behavior

 warning S1172: Remove this parameter 'x', whose value is ignored in the method.
 warning S1172: Remove this parameter 'y', whose value is ignored in the method.

### Related information

SonarAnalyzer.CSharp.8.6.0.16497
VisualStudio 2019 16.5.4

**Comment by pavel-mikula-sonarsource** [comment]
Hi @volgunin ,

Thank you for reporting this issue. We already have similar case reported in #3255 so I'll close this one.
